docs/4.2/gcp_guide.md

```diff
@@ -0,0 +1,203 @@
+# Running Teleport on GCP
+
+We've created this guide to give customers a high level overview of how to use Teleport
+on [Google Cloud](https://cloud.google.com/gcp/) (GCP). This guide provides a high level introduction leading to
+a deep dive into how to setup and run Teleport in production.
+
+We have split this guide into:
+
+- [Teleport on GCP FAQ](#teleport-on-gcp-faq)
+- [GCP Teleport Introduction](#gcp-teleport-introduction)
+- [GCP Quickstart](#gcp-quickstart)
+
+## Teleport on GCP FAQ
+#### Why would you want to use Teleport with GCP?
+As leader in [Beyond Corp](https://cloud.google.com/beyondcorp/), GCP already provides
+some great tools out of the box such as [Cloud Identity - Aware Proxy](https://cloud.google.com/iap/).  
+This is an excellent tool to quickly get setup securely with GCP but it can become 
+complicated to integrate into existing workflows and complicated if you want to share 
+tool across clouds. 
+
+#### Which Services can I use Teleport with?
+
+You can use Teleport for all the services that you would SSH into. This guide is 
+focused on Google Compute Engine. In the future we'll plan to update on how to use 
+Teleport with Kubernetes engine. 
+
+## GCP Teleport Introduction
+
+This guide will cover how to setup, configure and run Teleport on GCP.
+
+GCP Services required to run Teleport in HA 
+
+ - [Compute Engine: VM Instances with Instance Groups](#compute-engine-vm-instances-with-instance-groups)
+ - [Computer Engine: Health Checks](#computer-engine-health-checks)
+ - [Storage: Cloud Firestore](#storage-cloud-firestore)
+ - [Storage: Google Cloud Storage](#storage-google-cloud-storage)
+ - [Network Services: Load Balancing](#network-services-load-balancing)
+ - [Network Services: Cloud DNS](#network-services-cloud-dns)
+ 
+Other things needed
+
+ - [SSL Certificate ](https://cloud.google.com/load-balancing/docs/ssl-certificates)
+
+Optional:
+
+ - Management Tools: Cloud Deployment Manager
+ - Stackdriver Logging
+
+We recommend setting up Teleport in high availability mode (HA). In HA mode Firestore 
+stores the state of the system and Google Cloud Storage stores the audit logs.
+
+![GCP Intro Image](img/gcp/gcp-teleport.svg)
+
+
+### Compute Engine: VM Instances with Instance Groups
+To run Teleport in a HA configuration we recommend using `n1-standard-2` instances in Production. It's best practice to separate the proxy and authentication server, using Instance groups for the proxy and auth server. 
+
+### Computer Engine: Health Checks
+GCP relies heavily on [Health Checks](https://cloud.google.com/load-balancing/docs/health-checks), this is helpful when adding new instances to an 
+instance group. 
+
+To enable health checks in Teleport start with `teleport start --diag-addr=127.0.0.1:3000` see  [Admin Guide: Troubleshooting](admin-guide.md#troubleshooting) for more information. 
```

Comment:
In all likelihood you would need to configure Teleport to listen on `0.0.0.0` so that the load balancer healthcheck IP can access the instance. You'd then also need to add some kind of firewall rule to prevent anyone external from accessing the endpoint, but still allow the health checks.

Instructions on how to actually set up the healthchecks would be good too. (use HTTPS)

Reply:
I agree, I'll come back and update this. It tripped me up the first few time due to the manor in which GCP Load Balancers are configured.